flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -1463,21 +1465,23 @@ class CodeGenerator(
     */
   def addReusableFunction(function: UserDefinedFunction): String = {
     val classQualifier = function.getClass.getCanonicalName
-    val fieldTerm = s"function_${classQualifier.replace('.', '$')}"
+    val functionSerializedData = serialize(function)
+    val fieldTerm =
+      s"""
+         |function_${classQualifier.replace('.', '$')}_${DigestUtils.md5Hex(functionSerializedData)}
```

Comment:
1. You write the above two functions are stateless, there is no problem when create a instanse .
2. In fact, we do not have to worried about the variable name, we build a tree, each UDTF node is independent of the codegen.
So，we can keep named the `fieldTerm = function_${classQualifier.replace('.', '$')}`.